### ISSUE TITLE
Live TV: channel changing script invoked twice when typing channel number

### ISSUE THREAD

**Issue by JanCeuleers** [MythTV/mythtv#622]
* **Platform**: Ubuntu 20.04
Linux dracor 5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

* **MythTV version**: fixes/32
MythTV Version : v32.0+fixes.202208020504.cf553e438c~ubuntu20.04.1
MythTV Branch : fixes/32
Network Protocol : 91
Library API : 32.20200101-1
QT Version : 5.12.8
Options compiled in:
 linux profile use_hidesyms using_alsa using_oss using_pulse using_pulseoutput using_backend using_bindings_perl using_bindings_python using_bindings_php using_dvb using_firewire using_frontend using_hdhomerun using_satip using_vbox using_ceton using_joystick_menu using_libcec using_libcrypto using_gnutls using_libdns_sd using_libfftw3 using_libxml2 using_lirc using_mheg using_opengl using_egl using_qtwebkit using_qtscript using_qtdbus using_taglib using_v4l2 using_v4l2prime using_x11 using_system_libbluray using_systemd_notify using_systemd_journal using_drm using_bindings_perl using_bindings_python using_bindings_php using_freetype2 using_mythtranscode using_opengl using_egl using_drm using_vaapi using_nvdec using_vdpau using_ffmpeg_threads using_mheg using_libass using_libxml2 using_libmp3lame

* **Package version**: 
# apt-cache policy mythtv
mythtv:
  Installed: 2:32.0+fixes.202208020504.cf553e438c~ubuntu20.04.1
  Candidate: 2:32.0+fixes.202208020504.cf553e438c~ubuntu20.04.1
  Version table:
 *** 2:32.0+fixes.202208020504.cf553e438c~ubuntu20.04.1 500
        500 http://ppa.launchpad.net/mythbuntu/32/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/mythbuntu/32/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
     2:32.0+fixes.202204231836.daa4e7e447~ubuntu20.04.1 500
        500 http://ppa.launchpad.net/mythbuntu/32/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/mythbuntu/32/ubuntu focal/main i386 Packages
     2:31.0+fixes.20200323.9579662cdc-0ubuntu1 500
        500 http://be.archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages
        500 http://be.archive.ubuntu.com/ubuntu focal/multiverse i386 Packages

* **Component**: backend

### What steps will reproduce the bug?
1. Enter Live TV and select a source/input that is configured with a channel changing script. In my case this is a pvrusb2 tuner connected to the analogue outputs of an STB which needs its channels changing by means of IR.
2. Navigate to another channel by typing its channel number followed by enter.
3. Observe that the channel changing script is called twice, such that the channel number is blasted to the STB twice.

### How often does it reproduce? Is there a required condition?
This is 100% reproducible on my system.

### What is the expected behaviour?
The channel changing script should be called only once when changing channels.
In other channel changing scenarios, such as when starting a recording, or in Live TV when selecting another channel from the programme guide, the channel changing script is indeed invoked only once.
EDIT: Actually, I have now observed this issue when at the beginning of a recording as well.

### What do you see instead?
The channel changing script is invoked twice when the channel is changed in Live TV by typing the channel number and pressing enter. My STB can accept channel numbers having 1, 2 or 3 digits, so emitting the channel number twice is problematic for channel numbers that have only 1 or 2 digits (e.g. typing 1 enter results in channel 11, typing 23 results in a change to channel 232 immediately followed by a change to channel 3, etc.).

### Additional information
I am not a frequent Live TV user, so I can't be 100% sure, but I believe that this bug was not present in 31/fixes from which I recently upgraded.

**Comment by jhoyt4** [comment]
Possibly related or a duplicate of[ Issue 584](https://github.com/MythTV/mythtv/issues/584)

**Comment by JanCeuleers** [comment]
I can no longer reproduce this issue in 32, so I propose to close.
